src/main/java/ch/njol/skript/expressions/ExprExplodedBlocks.java

```diff
@@ -62,8 +86,89 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 	@Nullable
 	@Override
 	protected Block[] get(Event e) {
-		List<Block> blockList = ((EntityExplodeEvent) e).blockList();
-		return blockList.toArray(new Block[blockList.size()]);
+		return getBlocks(e).toArray(new Block[0]);
+	}
+
+	@Nullable
+	@Override
+	public Iterator<? extends Block> iterator(Event e) {
+		List<Block> blocks = getBlocks(e);
+		return new ArrayList(blocks).iterator();
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(final ChangeMode mode) {
+		switch (mode) {
+			case ADD:
+			case SET:
+			case DELETE:
+				return CollectionUtils.array(Block[].class);
+			case REMOVE:
+			case REMOVE_ALL:
+				return CollectionUtils.array(Block[].class, ItemType[].class);
+			case RESET:
+			default:
+				return null;
+		}
+	}
+	
+	@Override
+	public void change(Event e, final @Nullable Object[] delta, final ChangeMode mode) {
+		List<Block> blocks = getBlocks(e);
+
+		switch (mode) {
+			case DELETE:
+			case SET:
+			case ADD:
+				if (mode != ChangeMode.DELETE)
+					assert delta != null;
+				
+				if (mode != ChangeMode.ADD)
+					blocks.clear();
+
+				for (Object o : delta) {
+					if (o instanceof ItemType)
+						continue;
+
+					if (((Block) o).getType() != Material.AIR) // Performance
+						blocks.add((Block) o);
+				}
```

Comment:
```suggestion
				if (mode != ChangeMode.DELETE)
					for (Object o : delta) {
						if (o instanceof ItemType)
							continue;

						if (((Block) o).getType() != Material.AIR) // Performance
							blocks.add((Block) o);
					}
```
because delta can be null